FirebaseLab/app/src/main/java/com/ezequielc/firebaselab/MainActivity.java

```diff
@@ -0,0 +1,111 @@
+package com.ezequielc.firebaselab;
+
+import android.provider.Settings;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import com.firebase.ui.database.FirebaseRecyclerAdapter;
+import com.google.firebase.database.ChildEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+public class MainActivity extends AppCompatActivity {
+    private RecyclerView mRecyclerView;
+    private EditText mEditText;
+    private Button mButton;
+    private String mUsername;
+    private FirebaseRecyclerAdapter mAdapter;
+    private ArrayList<ChatMessage> mRoomChat;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        Random rng = new Random();
+        mRoomChat = new ArrayList<>();
+
+        String android_id = Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID);
+        if(android_id.equals( "669ec9813b4c140")){
+            mUsername = "User"+ rng.nextInt(499);
+        }else if (android_id.equals("53c3af768086e15e")){
```

Comment:
Why are you looking at specific device ID values? I'm guessing this was specific to your two phones for testing, but just to be clear, this isn't something you'd want to leave in production code.